pkg/querier/frontend/split_by_day_test.go

```diff
@@ -23,6 +23,11 @@ func TestNextDayBoundary(t *testing.T) {
 		{0, 1, millisecondPerDay - 1},
 		{0, 15 * seconds, millisecondPerDay - 15*seconds},
 		{1 * seconds, 15 * seconds, millisecondPerDay - (15-1)*seconds},
+		{14 * seconds, 15 * seconds, millisecondPerDay - (15-14)*seconds},
+		{0, 35 * seconds, millisecondPerDay - 20*seconds},
```

Comment:
I initially didn't understand the significance of the 20s here - a commend would be helpful. (As discussed on slack).